lighthouse/classes/event_properties/interfaces.py

```diff
@@ -0,0 +1,225 @@
+        """
+        ...
+
+    @abstractmethod
+    def value(self) -> Any:
+        """
+        Returns the value for the property. If the value cannot be obtained or
+        does not have a valid value, then it raises an exception.
+        NB: To avoid this it, the instance should be checked first with the
+        validate() method.
+
+        Arguments:
+            None
+
+        Returns:
+            Any - Value of the property
+
+        """
+        ...
+
+    @abstractmethod
+    def add_to_warehouse_message(self, message: WarehouseMessage) -> None:
+        """
+        Adds this event property information into the warehouse message
+
+        Arguments:
+            message: WarehouseMessage - A building message where we want to write
+            the information from this event property
+
+        Returns:
+            None
+
+        """
+        ...
+
+
+class EventPropertyAbstract(EventPropertyInterface):
+    """
+    This class provide some tools to validate data obtained directly
+    from the params argument.
+    It is an abstract class that will need to define the methods specified
+    in its interface class (EventPropertyInterface).
+    If a more advanced process is needed (like accessing an external service),
+    then the methods of this class should be rewritten (or a new class
+    should be created).
+    This class also contains methods that will be needed for internal
+    process for the class but should not be called from outside of this
+    instance (they should be considered all protected methods).
+    The public interface that should be used in this case is
+    defined in EventPropertyInterface.
+    """
+
+    def __init__(self, params: Dict[str, Any]):
+        """
+        Stores the params, resets the instance and perform a validation check on the params
+
+        Arguments:
+            params: Dict[str, Any] - dict of key/value with the settings for
+            this property
+        """
+        self.reset()
+        self._params = params
+        if self._params is None:
+            raise ValidationError("You need to define params to create the EventProperty")
+
+    def reset(self):
+        """
+        Deletes all errors and clears the validation setting.
+        """
+        self._errors: List[str] = []
+        self._value = None
+        self._validate = True
+
+    @property
+    def errors(self) -> List[str]:
+        """
+        Validates the instance and returns the complete list of errors found (this
+        errors are not just validations; it could also be previous exceptions
+        thrown during the lifetime of this instance).
+
+        Arguments:
+            None
+
+        Returns:
+            List[str] - List of error messages found currently
+        """
+        self.validate()
+        return self._errors
+
+    def valid(self) -> bool:
+        """Alias for #validate()"""
+
+        return self.validate()
+
+    def enforce_validation(self) -> None:
```

Comment:
validate() (is_valid()) performs a check of all available data before making any requests. It is a safe way of checking that we have everything we need before we actual try fetching the data, so validate() won't raise an exception (and if it does internally it will return false); it should be a safe method for the programmer using the class; I added also a comment trying to explain that in the comments of the EventPropertyInterface class.